kernel/comps/systree/src/tree.rs

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Defines the main `SysTree` structure and its root node implementation.
+
+use alloc::{
+    collections::BTreeMap,
+    string::{String, ToString},
+    sync::{Arc, Weak},
+    vec::Vec,
+};
+use core::any::Any;
+
+use ostd::mm::{VmReader, VmWriter};
+use spin::RwLock;
+
+use super::{
+    attr::SysAttrSet,
+    node::{SysBranchNode, SysNode, SysNodeId, SysNodeType, SysObj},
+    Error, Result, SysStr,
+};
+
+// --- SysTree Singleton Container ---
+
+/// The main container for the system tree.
+#[derive(Debug)]
+pub struct SysTree {
+    root: Arc<RootNode>,
+    // event_hub: SysEventHub,
+}
+
+impl SysTree {
+    /// Creates a new `SysTree` instance with a default root node
+    /// and standard subdirectories like "devices", "block", "kernel".
+    /// This is intended to be called once for the singleton.
+    pub(crate) fn new() -> Self {
+        let root_node = Arc::new(RootNode {
+            id: SysNodeId::new(),
+            name: "".into(),
+            attrs: SysAttrSet::new(),
+            children: RwLock::new(BTreeMap::new()),
+        });
+
+        // Add standard directories (using the same RootNode struct for simplicity for now)
+        // TODO: Implement proper directory node types if needed.
+        let devices_node = Arc::new(RootNode {
```

Comment:
Do not built any nodes into the root of `SysTree`. The `SysTree` is the mechanism part; the policy part should be the job of individual subsystems, e.g., the bus subsystem and the block subsystem, which register and manage the their nodes in `SysTree`. So do not add "device", "bus", or "block" nodes here.

Instead, to show that the `SysTree` works, add some basic unit tests. To ensure that these unit tests are run in CI, remember adding the `systree` crate to the `OSDK_CRATES` variable in the root `Makefile`.